lib/Models/WebMapServiceCatalogItem.js

```diff
@@ -1143,6 +1156,40 @@ WebMapServiceCatalogItem.prototype.exportData = function() {
     callWebCoverageService(this);
 };
 
+WebMapServiceCatalogItem.prototype.searchDataForMatchingFeatures = function(searchText, regex) {
+    const that = this;
+    return new Promise(function(resolve, reject) {
+        let workbenchItemResults = [];
+        if (that.hasLinkedWfs === false) {
+           resolve(workbenchItemResults);
+        } else if (that.hasLinkedWfs) {
+            that.linkedWfs.searchDataForMatchingFeatures(searchText, regex).then(function (res) {
+                workbenchItemResults = convertWfsSearchResultsToWms(res, that);
+                resolve(workbenchItemResults);
+            });
+        } else if (!defined(that.hasLinkedWfs)) {
+            loadXML(that.getCapabilitiesUrl.replace(/wms/gi, 'wfs'))
```

Comment:
Instead, return this promise. An important rule is: you must either handle (.then _and_ .catch) or return every promise.

Reply:
Also, we can potentially use `DescribeLayer` to discover WFS services.